src/utils/io.rs

```diff
@@ -102,4 +121,23 @@ where
         }
         Ok(())
     }
+
+    fn append_stack_sync(&mut self, dir: &PathBuf) -> io::Result<()> {
+        let dir = std::fs::read_dir(dir)?;
+
+        for entry in dir.flatten() {
+            let name = entry.file_name().to_string_lossy().to_string();
+            if !(self.filter)(&name) {
+                continue;
+            }
+            let entry_path = entry.path();
+            let metadata = entry.metadata();
+            match metadata {
```

Comment:
Why match here? Can we use `metadata()?;`? Or is it so that we check the rest of the dirs if we have a failture? If that's the case, it should be reflected in async version as well and we can simply do `continue` instead of pushing to stack right?

Reply:
This is exactly why, I wanted to continue with the rest of the dirs but with no metadata for one.

I can change the async version so that it has the same behaviour or I can return directly like in the async version.

Any preferences ?

Reply:
How does node handle this? For example if it can't access a subdirectory, does it throw error or return all other? I suspect the sooner.

Reply:
After looking at it more and this doc : https://man7.org/linux/man-pages/man3/readdir.3.html, I will fail fast instead of ignoring the metadata for one directory